doc/supervised.rst

```diff
@@ -94,17 +106,48 @@ This is useful because `pair_score` matches the **score** semantic of
 scikit-learn's `Classification metrics
 <https://scikit-learn.org/stable/modules/model_evaluation.html#classification-metrics>`_.
 
+For similarity learners `pair_distance` is not available, as they don't learn
+a distance. Intead you use `pair_score` that has the same behaviour but
+for similarity.
```

Comment:
based on the suggested organization above, this sentence should not be needed; you can simply re-use the warning you have below

Reply:
Removed and simplified.